server/src/test/java/org/elasticsearch/index/mapper/IpScriptFieldTypeTests.java

```diff
@@ -228,9 +229,15 @@ protected Query randomTermQuery(MappedFieldType ft, SearchExecutionContext ctx)
     }
 
     @Override
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/106900")
     public void testTermsQuery() throws IOException {
-        try (Directory directory = newDirectory(); RandomIndexWriter iw = new RandomIndexWriter(random(), directory)) {
+        try (
+            Directory directory = newDirectory();
+            RandomIndexWriter iw = new RandomIndexWriter(
+                random(),
+                directory,
+                LuceneTestCase.newIndexWriterConfig().setMergePolicy(LuceneTestCase.newMergePolicy(random(), false))
+            );
+        ) {
```

Comment:
This fix doesn't make sense to me. The result checking doesn't care about document order at all. 

So, I tested this fix running it over a couple thousand of iterations, and it still fails with the same assertion tripping. I am not 100% sure how to fix this test, but it doesn't have to do with the merge policy.

Reply:
Thansk for checking - I had checked for some hundred iterations and didn't hit a failure.

It's interesting that the change fixed the specific instance though. I'm not sure how to dig deeper into this.

Reply:
> It's interesting that the change fixed the specific instance though. I'm not sure how to dig deeper into this.

It could be because the random seed generation switched as different values were created. For example (note, my numbers here are fake), in the failing seed, only 4 integers would be created vs. 3 in your version. So, the values that cause the failure wouldn't actually be tripped.